src/Adapter/MSTest.CoreAdapter/ObjectModel/TestMethod.cs

```diff
@@ -44,6 +47,20 @@ public TestMethod(string name, string fullClassName, string assemblyName,  bool
             this.IsAsync = isAsync;
         }
 
+        public TestMethod(MethodBase method, string name, string fullClassName, string assemblyName, bool isAsync)
+            : this(name, fullClassName, assemblyName, isAsync)
+        {
+            if (method == null)
+            {
+                throw new ArgumentNullException(nameof(method));
+            }
+
+            ManagedNameHelper.GetManagedName(method, out var managedType, out var managedMethod);
```

Comment:
@Haplois Same comment here. `ManagedNameHelper` will not be available in the TP OM that ships with older versions of VS.